### PR TITLE
Update Jest Coverage Configuration

### LINKED ISSUE #108: Use collectCoverageFrom
Our calls to `jest` with `--coverage` don't fail for untested files. We should use `--collectCoverageFrom` to specify the directories and files we expect to be covered with each test command.

See: https://jestjs.io/docs/en/configuration#collectcoveragefrom-array

### PR DESCRIPTION
**Pull request by dhurlburtusa**
Update Jest coverage configuration to catch new files. Added coverage reporting to integration and e2e tests.

New tests added to complete unit test coverage.

Ignored some files for integration and e2e to get coverage at 100% for the files being tested.

Resolves #108 

## How has this been tested?

Ran `npm run verify`. Can still new coverage reports for integration and e2e tests. All three test types have 100% coverage due to some files being ignored.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,23 +111,37 @@ This will open a browser window of the Storybook webapp.
 
 Runs the unit tests with Jest.
 
-The unit tests are found throughout the project excluding tests in the `__tests__` directory which is being reserved for integration tests.
+The unit tests are found throughout the project excluding tests in the `__tests__` directory which is being reserved for integration tests and the `e2e` directory which is being reserved for e2e tests.
 
 ```sh
 npm run test:unit
 npm run test:unit -- --coverage
```
Comment (dhurlburtusa) [inline README.md:118]: Would simplify documentation and configuration if we always run coverage reports.

File: api/utils.ts [hunk 2]
```diff
@@ -16,9 +16,11 @@ const PRODUCTION_API_URL_ROOT = 'https://public-api.wordpress.com/wpcom/v2/exper
  * @throws UnauthorizedError
  */
 async function fetchApi(method: string, path: string, body: ApiDataSource | null = null) {
+  /* istanbul ignore next; code branch not reachable in integration tests -- we don't hit production */
```
Comment (dhurlburtusa) [inline api/utils.ts:19]: I think we can avoid these two `istanbul ignores` if the API_URL_ROOT is determined at build time.

File: jest.integration.config.js [hunk 3]
```diff
@@ -1,4 +1,18 @@
 module.exports = {
+  collectCoverageFrom: [
+    '<rootDir>/api/**/*.ts',
+    '!**/node_modules/**',
+    // FIXME: TODO: Get tests on the following.
+    '!<rootDir>/api/UnauthorizedError.ts',
```
Comment (dhurlburtusa) [inline jest.integration.config.js:6]: This is only thrown in the branch of code that only runs on production. We don't plan to test production with integration tests. So, `UnauthorizedError.ts` may be permanently ignored. If so, we can add a comment and remove the `FIXME` and `TODO` comment. Thoughts?

File: jest.e2e.config.js [hunk 4]
```diff
@@ -1,4 +1,23 @@
 module.exports = {
+  collectCoverageFrom: [
+    '<rootDir>/pages/**/*.{ts,tsx}',
+    '!**/node_modules/**',
+    // FIXME: TODO: Get tests on the following.
+    '!<rootDir>/pages/_app.tsx',
+    '!<rootDir>/pages/auth.tsx',
+    '!<rootDir>/pages/index.tsx',
+    '!<rootDir>/pages/experiments/[id].tsx',
+    '!<rootDir>/pages/experiments/[id]/results.tsx',
```
Comment (dhurlburtusa) [inline jest.e2e.config.js:10]: Should we address these now or the final (or is it Week 0?) week of the milestone?

File: README.md (hunk 1 above)
Comment (yanirs) [reply README.md:118]: Yeah, I think it makes sense to always run coverage reports.

File: jest.e2e.config.js (hunk 4 above)
Comment (yanirs) [reply jest.e2e.config.js:10]: Let's add an issue to address these when the pages are more stable.

File: jest.integration.config.js (hunk 3 above)
Comment (yanirs) [reply jest.integration.config.js:6]: Can note that this is related to #52 and #64. We should be able to cover it if we test against the production API or a more sophisticated mock.

File: jest.e2e.config.js (hunk 4 above)
Comment (dhurlburtusa) [reply jest.e2e.config.js:10]: Done.

**Pull request merged**